app/controllers/account/pages_controller.rb

```diff
@@ -1,14 +1,17 @@
 class Account::PagesController < ApplicationController
   def index
-    @pages = collection.page(params[:page]).per(10)
+    @pages   = collection.page(params[:page]).per(10)
+    @company = @pages.first.company
```

Comment:
BUG

what if @pages = []

Reply:
find @company from params.
pages are nested and you have company_id in params

Reply:
i'd change a `#collection` a bit and added additional method to find company, e.g.:
```
def company_resource
  @company_resource ||= current_user.companies.find(params[:company_id])
end

def collection
  company_resource.pages.by_rating
end
```

the rest implementation is on you

Reply:
you can go even more abstract with

```
def parent
  @company ||= current_user.companies.find(params[:company_id])
end

def collection
  parent.pages # I also suggest to return collection without scopes and then add scopes in "index" itself
end
```